### PR TITLE
Feature/use case 6

### PR DESCRIPTION
**Pull request by Craig180885-napier**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -47,12 +47,15 @@
                         <groupId>org.junit.jupiter</groupId>
                         <artifactId>junit-jupiter-engine</artifactId>
                         <version>[5.1.0,)</version>
-                    </dependency>
+                      </dependency>
```
Comment (LukasP23) [inline pom.xml:50]: Remove this space immediately!

File: src/main/java/com/napier/sem/App.java [hunk 2]
```diff
@@ -85,14 +85,23 @@ public static void main(String[] args) {
         // Produce a report on all capital cities in a region organised by largest population to smallest
         // ArrayList<Record> capitalCities = dao.allCapitalCitiesIn(REGION, "Caribbean");
 
+
+        // Use-Cases 6.1-6.3
+        // 6.1 - Produce a report on the top N capital cities in the world organised by largest population to smallest
+        // ArrayList<Record> capitalCities = dao.topNCapitalCitiesIn(WORLD, "",5);
+        // 6.2 - Produce a report on the top N capital cities in the world organised by largest population to smallest
+        // ArrayList<Record> capitalCities = dao.topNCapitalCitiesIn(CONTINENT, "Asia", 5);
+        // 6.3 - Produce a report on the top N capital cities in the world organised by largest population to smallest
```
Comment (LukasP23) [inline src/main/java/com/napier/sem/App.java]: I assume you've forgotten to change these comments?

File: src/main/java/com/napier/sem/DAO.java [hunk 3]
```diff
@@ -194,8 +194,7 @@ public static String getWhereCondition(String areaFilter, String areaName)
      *
      * @return An ordered list of capital cities in a specific area sorted by descending population
      */
-    public ArrayList<Record> allCapitalCitiesIn(String areaFilter, String areaName)
-    {
+    public ArrayList<Record> allCapitalCitiesIn(String areaFilter, String areaName) {
```
Comment (LukasP23) [inline src/main/java/com/napier/sem/DAO.java:197]: Move the bracket onto the next line!

File: src/main/java/com/napier/sem/DAO.java [hunk 4]
```diff
@@ -208,4 +207,27 @@ public static String getWhereCondition(String areaFilter, String areaName)
...
         return executeStatement(statementString, App.CAPITAL_CITY);
     }
+
+    /**
+     * Use cases 6.1-6.3
+     * Constructs an SQL query to fetch the top N populated cities in a specific area, and executes the query.
+     *
+     * @return An ordered list of cities in a defined area sorted by descending population
+     */
+    public ArrayList<Record> topNCapitalCitiesIn(String areaFilter, String areaName, Integer n) {
```
Comment (LukasP23) [inline src/main/java/com/napier/sem/DAO.java:217]: Opening bracket on new line!!

File: src/main/java/com/napier/sem/App.java (hunk 2 above)
Comment (scan-lan) [inline src/main/java/com/napier/sem/App.java]: these comments need to be updated to reflect the actual use cases, e.g. top N capital cities in a continent organised by largest population to smallest

File: src/main/java/com/napier/sem/Record.java [hunk 5]
```diff
@@ -46,7 +47,6 @@ public Record(ResultSet result, String recordType) throws SQLException
                 region = result.getString("region");
                 population = result.getInt("population");
                 capital = result.getString("capital");
-                this.recordType = recordType;
```
Comment (LukasP23) [inline src/main/java/com/napier/sem/Record.java:49]: Why have you removed this? I thought the issue was that it hadn't been added to each case

Comment (LukasP23) [reply src/main/java/com/napier/sem/Record.java:49]: Nevermind I see it

File: src/main/java/com/napier/sem/DAO.java (hunk 3 above)
Comment (scan-lan) [inline src/main/java/com/napier/sem/DAO.java:197]: could we make this a drop bracket?

File: src/main/java/com/napier/sem/DAO.java [hunk 6]
```diff
@@ -208,4 +207,27 @@ public static String getWhereCondition(String areaFilter, String areaName)
...
+        String whereCondition = getWhereCondition(areaFilter, areaName);
+
+
+        // Define the SQL query as a string
+        String statementString = "SELECT city.name, city.population, country.region, country.continent, country.name AS country \n" +
+                "FROM city\n" +
+                "    JOIN country ON city.countrycode = country.code\n" +
+                "WHERE city.id = country.capital \n" +
```
Comment (scan-lan) [inline src/main/java/com/napier/sem/DAO.java]: this `WHERE` could be part of the join, which might be a little more efficient. i'm happy leaving it like this, or i can should you how i'd change it.

**Review by scan-lan — DISMISSED** [review]

File: pom.xml (hunk 1 above)
Comment (Craig180885-napier) [reply pom.xml:50]: fuck off

File: src/main/java/com/napier/sem/DAO.java (hunk 3 above)
Comment (Craig180885-napier) [reply src/main/java/com/napier/sem/DAO.java:197]: intellij reformats on it's own

File: src/main/java/com/napier/sem/DAO.java (hunk 4 above)
Comment (Craig180885-napier) [reply src/main/java/com/napier/sem/DAO.java:217]: intellij reformats on it's own but I'll try

**Review by LukasP23 — APPROVED** [review]

**Pull request merged**